### ISSUE TITLE
Worksheet.RangeUsed( true ) ignores a cell with value

### ISSUE THREAD

**Issue by crypto-rsa** [ClosedXML/ClosedXML#881]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**
e.g. 0.93-beta4 (tested also in the `develop` branch)

**What is the current behavior?**
Calling `Worksheet.RangeUsed( true )` returns a range which doesn't include a non-empty cell.

**What is the expected behavior or new feature?**
`Worksheet.RangeUsed` (regardless of the value of the `includeFormats` parameter) should return a range which includes all cells which have a value.

**Did this work in previous versions of our tool?  Which versions?**
Don't know.

## Reproducibility
**Code to reproduce problem:**
```c#
public void Main()
{
	using( var wb = new XLWorkbook( "Test.xlsx" ) )
	{
		var ws = wb.Worksheets.First();
		Console.WriteLine( ws.RangeUsed( false ).ToString() );
		Console.WriteLine( ws.RangeUsed( true ).ToString() );
	}
}
```

This results in
```
Test!A1:B2
Test!B2:D2
```

The second call ignores cell A1.

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[Test.xlsx](https://github.com/ClosedXML/ClosedXML/files/2053878/Test.xlsx)

**Comment by igitur** [comment]
Did this work in 0.92?

**Comment by Pankraty** [comment]
I didn't test but from what I see in code - I really doubt it worked.

Question: now we consider merged ranges only if includeFormats is true. I think we should analyze them in both cases.

**Comment by Pankraty** [comment]
Another problem: `LastCellUsed` and `FirstCellUsed` ignore predicates when analyzing merged ranges. This is wrong too.

**Comment by igitur** [comment]
My comment was meant for @crypto-rsa .

@crypto-rsa Please test whether this worked on 0.92.

If this is a regression, it should be fixed for the 0.93 release. If not, it can wait.

**Comment by crypto-rsa** [comment]
> Did this work in 0.92?
No.
